saleor/core/extensions/manager.py

```diff
@@ -0,0 +1,181 @@
+import importlib
+from typing import TYPE_CHECKING, Any, List, Union
+
+from django.conf import settings
+from django_countries.fields import Country
+from prices import Money, MoneyRange, TaxedMoney, TaxedMoneyRange
+
+from ..taxes import TaxType, quantize_price
+
+if TYPE_CHECKING:
+    from ...checkout.models import Checkout, CheckoutLine
+    from ...product.models import Product
+    from ...account.models import Address
+    from ...order.models import OrderLine, Order
+
+
+class BaseManager:
+    """Base manager for handling a plugins logic"""
+
+    plugins = None
+
+    def __init__(self, plugins: List[str]):
+        self.plugins = []
+        for plugin_path in plugins:
+            plugin_path, _, plugin_name = plugin_path.rpartition(".")
+            plugin_module = importlib.import_module(plugin_path)
+            plugin_class = getattr(plugin_module, plugin_name)
+            self.plugins.append(plugin_class())
+
+    def run_plugin_method(self, name: str, default_value: Any, *args, **kwargs):
+        value = default_value
+        for p in self.plugins:
+            plugin_value = getattr(p, name, NotImplemented)(
```

Comment:
This is extremely unclear what is happening here. Can we have some kind of function extracted from here with proper name or at least other kind of explanation? Impossible to quickly reason about types being passed around here.

Reply:
Logic has been split to improve readability